app/commands/solution/index.rb

```diff
@@ -0,0 +1,65 @@
+class Solution::Index
```

Comment:
```suggestion
class Solution::SyncToSearchIndex
```

As `index` can be a verb or a noun, I initially read this as being a "thing", not a command, so I think I'd rather rename to be more explicit.